sdk/core/Azure.Core/api/Azure.Core.net461.cs

```diff
@@ -927,6 +931,12 @@ public partial class ServerCertificateCustomValidationArgs
         public System.Security.Cryptography.X509Certificates.X509Chain? CertificateAuthorityChain { get { throw null; } }
         public System.Net.Security.SslPolicyErrors SslPolicyErrors { get { throw null; } }
     }
+    public partial class TelemetryPackageInfo
+    {
+        internal TelemetryPackageInfo() { }
+        public string UserAgentValue { get { throw null; } }
+        public static Azure.Core.Pipeline.TelemetryPackageInfo FromType<T>(string? applicationId = null) { throw null; }
```

Comment:
I worry about this method. It gives the impression that type and application is all you will ever need to create an instance of this type. I think option types should be simple property bags. If we need a routine to format UA strings, we should put it somewhere, e.g. on the policy.

Reply:
The type information is intentionally all that is needed to create an instance of this type. Its intent was not to have it be a property bag of options, since that is what the dictionary itself is. This type represents a specific option and value (even if we decide on a different name).

Reply:
If we don't plan to add anything else (other properties to the TelemetryPackageInfo type), then why do we need this type? Why not simply use a string instead using TelemetryPackageInfo as the value of the property?

Reply:
OK, I think I know the answer to my question above. I think what you are after is you want to prevent people from sticking random strings into the UA header and you want to avoid the cost of formatting on every call. If that's the reason, then what about:
```csharp
public static class HttpMessageExtensions {
    public static void SetUserAgent(this HttpMessage message, UserAgentString userAgent)
}

public class UserAgentString {
    private UserAgentString () {}
    public static UserAgentString Create(Assembly assembly, string applicationId);
    public override string ToString();
}
```

Reply:
If we only take the Assembly, we can get rid of the factory method and just have a constructor. It might be better to be more explicit about what the UserAgentString operates on, though I agree it is slightly less convenient for users.

Reply:
I prefer the convenient approach. I'll make the doc comments more explicit about how `Assembly` is derived from the `Type`